### DISCUSSION TITLE
Variants not recognized

### DISCUSSION THREAD

**Discussion by captDaylight** [tamagui/tamagui#183] [Q&A]
I grabbed some of Tamagui’s example code for variants [directly](https://tamagui.dev/docs/core/styled) from the documentation and it’s throwing errors:

![image](https://user-images.githubusercontent.com/700617/187779080-4763ef15-6731-45f3-a108-4dd845e7f772.png)

**Comment by ibedwi** [comment]
I also experienced the same issue. I think there's a problem with the returned component's type.

Tried to manually define the type and it's work:

```typescript
import { Stack, styled } from '@tamagui/core';

type PinType = "top" | "bottom" | "left"; 

type Variants = {
  pin: {
    [k in PinType]: {
      [key: string]: string | number | any;
    };
  };
  centered: {
     true: {
      [key: string]: string | number | any;
    };
  };
};

// Manually define the `ParentComponent` and `Variants` type.
// Ref: https://github.com/tamagui/tamagui/blob/master/packages/core/src/styled.tsx#L25
export const Circle = styled<typeof Stack, Variants>(Stack, {
  name: 'Circle',
  borderRadius: 100_000_000,
  backgroundColor: 'aqua',

  variants: {
    pin: {
      top: {
        position: 'absolute',
        top: 0,
      },
      bottom: {},
      left: {}
    },

    centered: {
      true: {
        alignItems: 'center',
        justifyContent: 'center',
      },
    },
  },
});
```

And if you I use the component, it worked:

<img width="418" alt="Screen Shot 2022-09-03 at 04 01 41" src="https://user-images.githubusercontent.com/15979404/188235461-cf1af86a-d6d7-4e01-96b7-e246811c3237.png">

**Reply by ibedwi** [reply]
Not sure, but I don't think we suppose to add the types manually 🤔

**Reply by ibedwi** [reply]
I've submitted an issue, please check [here](https://github.com/tamagui/tamagui/issues/184) 😁

**Comment by captDaylight** [comment]
Nice one @ibedwi! I'll track the issue 👍
